core/domain/classifier_services.py

```diff
@@ -214,6 +214,13 @@ def save_classifier(classifier):
         classifier_id = _create_classifier(classifier)
     else:
         _update_classifier(classifier_data_model, classifier.state_name)
+
+    # Create the mapping from <exp_id,exp_version,state_name>
+    # to classifier_id.
+    create_classifier_exploration_mapping(
```

Comment:
@seanlip @pranavsid98 @prasanna08 yep, the original intent was to update the classifiers when the exploration is updated (i.e. when the state_name changes). But now with the mapping, we don't need it. In fact, we don't need to store the state_name in the classifier_data anymore. If required, a look-up in the exploration-classifier mapping would give us the required state_name. However, to make it easier while debugging, I don't mind keeping the state_name around.

Also, it is better to keep create_classifier_exploration_mapping() inside save_classifier() because outside the save_classifier method, you shouldn't really care how a classifier is stored and what other steps you need to worry about.